huaweicloud/services/iec/resource_huaweicloud_iec_vpc.go

```diff
@@ -0,0 +1,149 @@
+package iec
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/iec/v1/vpcs"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func ResourceIecVpc() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIecVpcCreate,
+		ReadContext:   resourceIecVpcRead,
+		UpdateContext: resourceIecVpcUpdate,
+		DeleteContext: resourceIecVpcDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cidr": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "SYSTEM",
+			},
+			"subnet_num": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIecVpcCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	iecClient, err := conf.IECV1Client(conf.GetRegion(d))
+
+	if err != nil {
+		return diag.Errorf("error creating IEC client: %s", err)
+	}
+
+	createOpts := vpcs.CreateOpts{
+		Name: d.Get("name").(string),
+		Cidr: d.Get("cidr").(string),
+		Mode: d.Get("mode").(string),
+	}
+
+	n, err := vpcs.Create(iecClient, createOpts).Extract()
+	if err != nil {
+		return diag.Errorf("error creating IEC VPC: %s", err)
+	}
+
+	log.Printf("[INFO] IEC VPC ID: %s", n.ID)
+	d.SetId(n.ID)
+
+	return resourceIecVpcRead(ctx, d, meta)
+}
+
+func resourceIecVpcRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	iecClient, err := conf.IECV1Client(conf.GetRegion(d))
+	var mErr *multierror.Error
+	if err != nil {
```

Comment:
we should check the error as soon as possible, please check the same issue in vpc_subnet

```suggestion
	iecClient, err := conf.IECV1Client(conf.GetRegion(d))
	if err != nil {
```